src/platforms/gbm-kms/server/kms/quirks.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ */
+
+#include "quirks.h"
+
+#include "mir/log.h"
+#include "mir/options/option.h"
+#include "mir/udev/wrapper.h"
+
+#include <vector>
+#include <unordered_set>
+#include <boost/assign.hpp>
+
+namespace mgg = mir::graphics::gbm;
+namespace mo = mir::options;
+
+namespace
+{
+char const* quirks_option_name = "driver-quirks";
+}
+
+namespace
+{
+auto value_or(char const* maybe_null_string, char const* value_if_null) -> char const*
+{
+    if (maybe_null_string)
+    {
+        return maybe_null_string;
+    }
+    else
+    {
+        return value_if_null;
+    }
+}
+}
+
+class mgg::Quirks::Impl
+{
+public:
+    explicit Impl(mo::Option const& options)
+    {
+        if (!options.is_set(quirks_option_name))
+        {
+            return;
+        }
+
+        for (auto const& quirk : options.get<std::vector<std::string>>(quirks_option_name))
+        {
+            // Quirk format is skip:(devnode|driver):value
+
+            if (quirk.size() < 5)
+            {
+                mir::log_warning(
+                    "Ignoring unexpected value for %s option: %s (expects value of the form “skip:<type>:<value>”)",
+                    quirks_option_name,
+                    quirk.c_str());
+                continue;
+            }
+            if (quirk[0] != 's' ||
+                quirk[1] != 'k' ||
+                quirk[2] != 'i' ||
+                quirk[3] != 'p' ||
+                quirk[4] != ':')
```

Comment:
`strncmp()`?

Reply:
That would indeed be the function I was hunting for, thanks.